pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -17,30 +17,34 @@
 , openclSupport ? false
 , clblast
 
-, openblasSupport ? !rocmSupport
+, openblasSupport ? !rocmSupport && !cudaSupport
 , openblas
 , pkg-config
 }:
 
 assert lib.assertMsg
-  (lib.count lib.id [openclSupport openblasSupport rocmSupport] == 1)
-  "llama-cpp: exactly one  of openclSupport, openblasSupport and rocmSupport should be enabled";
+  (lib.count lib.id [openclSupport openblasSupport rocmSupport cudaSupport] == 1)
+  "llama-cpp: exactly one  of openclSupport, openblasSupport, cudaSupport and rocmSupport should be enabled";
```

Comment:
Better moved to `meta.broken = ...`. Assertions are sometimes justifiable but most of the time they just break evaluation and are hard to debug